### PR TITLE
FBXLoader: Check null textures before assignment

### PR DESCRIPTION
**Pull request by camnewnham**
**Description**

We've found that a variety of FBX exporters from CAD software fail to correctly create/assign textures. As a result, `scope.getTexture` returns null/undefined, but the code fails when attempting to assign an encoding or other property immediately after.

As such, the preferred behaviour would be to load what is possible from the model (sans textures if they are not able to load) rather than throw and exception.

### CODE REVIEW

**Review by looeee — APPROVED** [review]

**Comment by looeee** [comment]
This looks good, thanks! 👍

File: examples/jsm/loaders/FBXLoader.js [hunk 1]
```diff
@@ -620,7 +620,9 @@ class FBXTreeParser {
 				case 'DiffuseColor':
 				case 'Maya|TEX_color_map':
 					parameters.map = scope.getTexture( textureMap, child.ID );
-					parameters.map.encoding = sRGBEncoding;
+					if (parameters.map != null) {
```
Comment (Mugen87) [inline examples/jsm/loaders/FBXLoader.js:623]: Please use `!==.

**Comment by camnewnham** [comment]
I've updated this some more to not assign the `material.map` field if the texture (`material.map.image`) is missing. This is because exporters (GLTFExporter in my tests) check for material.map and assume that means an image is assigned. Would it be appropriate to `console.warn` here to notify the user/developer that the textures are missing?

**Comment by Mugen87** [comment]
I suggest you undo a1999fe3ccad3b5357a45bf2689d9136a84d5cfe. The previous commit a2219884d9f7343f04d0d8b22a55258d8770bfd8 already looked fine.

**Comment by camnewnham** [comment]
> I suggest you undo [a1999fe](https://github.com/mrdoob/three.js/commit/a1999fe3ccad3b5357a45bf2689d9136a84d5cfe). The previous commit [a221988](https://github.com/mrdoob/three.js/commit/a2219884d9f7343f04d0d8b22a55258d8770bfd8) already looked fine.

Without this, the imported material has map defined but map.image undefined - this causes issues with GLTFExporter (and perhaps others) which checks for map !== undefined and then uses map.image.

Would you prefer I reversed this commit and changed GLTFExporter to check for this instead?

**Comment by Mugen87** [comment]
No, it's better to fix this in `FBXLoader`.

Thanks for the additional explanation. Wouldn't it be better if `getTexture()` does not return an incomplete texture? In this way, the check for an undefined image has to be implemented only once. 

Besides, `if (bumpMap?.image) {` does not look like valid JS syntax.

**Comment by mrdoob** [comment]
It is valid actually:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Maybe we can start adopting it?
https://caniuse.com/mdn-javascript_operators_optional_chaining

**Comment by mrdoob** [comment]
But yes, adding this logic to `getTexture()` seems cleaner indeed.

**Comment by Mugen87** [comment]
Interesting! Support looks sufficiently good.

**Comment by camnewnham** [comment]
Thanks for the feedback. I've updated this to return undefined from `getTexture()` when the image is undefined.

**Comment by Mugen87** [comment]
Cleaned up the code style a bit...

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**